plugins/processors/t128_transform/t128_transform.go

```diff
@@ -78,6 +86,54 @@ type observedValue struct {
 	timestamp time.Time
 }
 
+func (o observedValue) MarshalJSON() ([]byte, error) {
+	value := struct {
+		Value        interface{}  `json:"value"`
+		Previous     interface{}  `json:"previous"`
+		Expires      string       `json:"expires"`
+		Timestamp    string       `json:"timestamp"`
+	}{
+		Value:        o.value,
+		Previous:     o.previous,
+		Expires:      o.expires.Format(time.RFC3339),
```

Comment:
I think `observedValue` is just missing the Public attribute names and then you wouldn't need this custom marshal/unmarshal code

Reply:
good one